apps/prairielearn/src/sync/course-db.ts

```diff
@@ -815,107 +772,105 @@ export async function loadCourseInfo(courseId, coursePath) {
 }
 
 /**
- * @template {{ uuid: string }} T
- * @param {Object} options - Options for loading and validating the file
- * @param {string} options.coursePath
- * @param {string} options.filePath
- * @param {any} options.defaults
- * @param {any} options.schema
+ * @param options
  * @param {boolean} [options.tolerateMissing] - Whether or not a missing file constitutes an error
```

Comment:
Unnecessary type information on this line.

Reply:
I have removed it in cfe1ee6dad3e00c831a5a34a1a80687905596df1. I had originally removed this, but had some problems with it. The parameter comments show up in VSCode when hovering over the function, but it does not give the help option when hovering over the parameter itself, so it's not making the connection between the @param comment and the named parameter itself.

Reply:
The correct way to provide documentation on properties of this kind of options object is to add a comment to the property itself. In this case, you'd have the following:

```ts
sync function loadAndValidateJson<T extends { uuid: string }>({
  coursePath,
  filePath,
  defaults,
  schema,
  validate,
  tolerateMissing,
}: {
  coursePath: string;
  filePath: string;
  defaults: any;
  schema: any;
  /** Whether or not a missing file constitutes an error. */
  tolerateMissing?: boolean;
  validate: (info: T) => Promise<{ warnings: string[]; errors: string[] }>;
}): Promise<InfoFile<T> | null> {
```

Reply:
Ok, I moved them in 5528e63f21088bae6d51ea0dcd4a24dac511e065. I had tried to add them to the variable name on top, I didn't try it with the type definition itself. It works now.